pkg/kubelet/dockertools/manager.go

```diff
@@ -1354,6 +1354,10 @@ func (dm *DockerManager) killContainer(containerID kubecontainer.ContainerID, co
 	if gracePeriod < minimumGracePeriodInSeconds {
 		gracePeriod = minimumGracePeriodInSeconds
 	}
+	if gracePeriodOverride != nil {
```

Comment:
Defaults are handled by Kube API.  This is the fundamental protection that
we always give EVERYONE at least 2s to respond to sigterm.  Even if you say
0s.

I'm fine for this to bypass grace period and kill directly.  I just wanted
to have the comment recorded.

On Fri, Apr 29, 2016 at 3:56 PM, Derek Carr <notifications@github.com>
wrote:

> In pkg/kubelet/dockertools/manager.go
> <https://github.com/kubernetes/kubernetes/pull/24843#discussion_r61635974>
> :
>
> > @@ -1354,6 +1354,10 @@ func (dm *DockerManager) killContainer(containerID kubecontainer.ContainerID, co
> >  	if gracePeriod < minimumGracePeriodInSeconds {
> >  		gracePeriod = minimumGracePeriodInSeconds
> >  	}
> > +	if gracePeriodOverride != nil {
>
> btw, systemd defaults to 90s for default stop timeout, not sure why we
> settled on 2s here...
>
> https://www.freedesktop.org/software/systemd/man/systemd-system.conf.html
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/24843/files/2f45241a8f7c3845dbf096d73b65ae046a0b71bf#r61635974>
>